### PR TITLE
Added docs for running GUI tests

### PR DESCRIPTION
**Pull request by swoichha**
### Description
This PR adds document on how to set up client and squish locally, and run GUI tests.

### Related Issue
https://github.com/owncloud/client/issues/8705

### CODE REVIEW

**Comment by mmattel** [comment]
@EParzefall could you please do a language check?

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+== Introduction
+
+This document explains how to run GUI tests for the Desktop Client locally in your system. To run GUI tests, the Squish GUI Test Automation Tool for all kinds of cross-platform desktop, mobile, embedded and web applications is used.
+
+== Prerequisites
+
+Before we can actually run the test, we will need to make a Desktop Client build and install and configure Squish first.
+
+NOTE: Before running tests, you need to make sure that you have disabled `oauth2` and `openidconnect` in owncloud server.
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:23]:
```suggestion
NOTE: Before running tests, you need to make sure that you have disabled `oauth2` and `openidconnect` in ownCloud Server.
```

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+=== Building the Desktop Client
+
+To be able to run tests, you need to xref:appendices/building.adoc[build the Desktop Client] from latest master.
+
+=== Install Squish
+
+After building the ownCloud Client, install and configure squish. There are two ways to install squish:
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:31]:
```suggestion
After building the ownCloud Client, install and configure squish. There are two ways to install Squish:
```

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+[source,console]
+----
+sudo ./squish-6.6.2-qt512x-linux64.run
+----
+. You will be asked for a license key. When asked, enter your existing license or get a {free-trial-url}[free trial license]
+. After you have entered the license key, Squish opens.
+
+==== Installing Squish Using Docker
+
+You can also use the {squish-docker-image-url}[squish docker image] to run tests. Just proceed with following steps:
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:59]:
```suggestion
You can also use the {squish-docker-image-url}[Squish docker image] to run tests. Proceed with the following steps:
```

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+Pull the docker image with following command:
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:61]:
```suggestion
Pull the docker image with the following command:
```

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+. Run the downloaded script.
++
+[source,console]
+----
+sudo python3 PythonChanger.py --force
+----
+
+== Prepare Tests
+
+Some necessary steps before running test.
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:83]:
```suggestion
Some necessary steps before running tests:
```

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+. Clone the {client-repo-url}[Desktop Client] from GitHub
+. Copy `test/gui/config.sample.ini` to `test/suite_oc-desktop/config.ini`
+. Edit `test/gui/config.ini` and set `BACKEND_HOST=` to the URL of your owncloud server e.g. `BACKEND_HOST=http://localhost/owncloud-core`
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:87]:
```suggestion
. Edit `test/gui/config.ini` and set `BACKEND_HOST=` to the URL of your ownCloud server, e.g. `BACKEND_HOST=http://localhost/owncloud-core`
```

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+. Start Squish
+. Open the existing test-suite via: menu:File[Open Test Suite > test/gui]
+. Go to menu:Edit[Server Settings > Manage AUTs > Mapped AUTs]
+. Click btn:[Add] and select the compiled binary, eg: `client/client-build/bin/owncloud`
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:91]:
```suggestion
. Click btn:[Add] and select the compiled binary, e.g.: `client/client-build/bin/owncloud`
```

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+. Close any running Desktop Client, by clicking btn:[Quit ownCloud] in the settings page
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:92]:
```suggestion
. Close any running Desktop Clients by clicking btn:[Quit ownCloud] in the settings page
```

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+. Run the AUT (Application under Test) to check if everything works properly via menu:Run[Lauch AUT] (the Desktop Client settings window should appear)
+
+== Run Rests
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:95]:
```suggestion
== Run Tests
```

**Review by EParzefall — APPROVED** [review]
LGTM

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+* xref:installing-squish-using-an-executable-file[Installing Squish Using an Executable File]
+* xref:installing-squish-using-docker[Installing Squish Using Docker]
+
+==== Installing Squish Using an Executable File
+
+To install Squish, follow these steps:
+
+. Download the latest version of Squish from {squish-url}[froglogic] to a location of your choice.
+. Depending upon the version and system you are using, you will get a file like `squish-6.6.2-qt512x-linux64.run`.
+. Make the downloaded file executable by runninng following example command:
```
Comment (phil-davis) [inline modules/ROOT/pages/appendices/guitest.adoc:42]:
```suggestion
. Make the downloaded file executable by running the following example command:
```

**Comment by mmattel** [comment]
@swoichha
do you plan more changes
and do we need backporting to 2.8 and 2.7 (else it will only go to master and any upcoming 2.9 release)

**Comment by swoichha** [comment]
> @swoichha
> do you plan more changes (because of additional language checks...)

@mmattel There are no changes to be added expect the language check. @EParzefall can you review the language once again? 

> do we need backporting to 2.8 and 2.7 (else it will only go to master and any upcoming 2.9 release)
Yes we need backporting to 2.8 and 2.7 as well.

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+NOTE: Before running tests, you need to make sure that you have disabled `oauth2` and `openidconnect` in ownCloud Server.
+
+=== Building the Desktop Client
+
+To be able to run tests, you need to xref:appendices/building.adoc[build the Desktop Client] from latest master.
+
+=== Install Squish
+
+After building the ownCloud Client, install and configure squish.
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:31]:
```suggestion
After building the ownCloud Client, install and configure Squish.
```

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+* xref:run-tests-using-docker[Run Tests Using Docker]
+
+=== Run Tests Using Squish
+
+* Start the {owncloud-test-middleware-url}[owncloud-test-middleware]
+* Click the play-button for a {test-case-scenario-url}[test-case or scenario].
+
+=== Run Tests Using Docker
+
+* Run squish docker image using the following command:
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:113]:
```suggestion
* Run the Squish docker image using the following command:
```

File: modules/ROOT/pages/appendices/guitest.adoc [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+
+== Run Tests
+
+* xref:run-tests-using-squish[Run Tests Using Squish]
+* xref:run-tests-using-docker[Run Tests Using Docker]
+
+=== Run Tests Using Squish
+
+* Start the {owncloud-test-middleware-url}[owncloud-test-middleware]
+* Click the play-button for a {test-case-scenario-url}[test-case or scenario].
```
Comment (EParzefall) [inline modules/ROOT/pages/appendices/guitest.adoc:109]:
```suggestion
* Click the play button for a {test-case-scenario-url}[test-case or scenario].
```

**Review by EParzefall — APPROVED** [review]
two minor fixes

**Pull request merged**